### PR TITLE
Stream_name: Adding a default stream name to the Stream Info.

### PR DESCRIPTION
**Pull request by PawBud**
This PR aims to fix https://bugs.launchpad.net/mixxx/+bug/1818054.

Discussion on Zulip: https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/adding.20a.20default.20Stream.20Name.

### CODE REVIEW

File: src/preferences/broadcastprofile.cpp [hunk 1]
```diff
@@ -76,6 +76,8 @@ const bool kDefaultNoDelayFirstReconnect = true;
 const bool kDefaultOggDynamicupdate = false;
 double kDefaultReconnectFirstDelay = 0.0;
 double kDefaultReconnectPeriod = 5.0;
+const QString kDefaultStreamName =
+        QObject::tr("Mixxx");
```
Comment (ronso0) [inline src/preferences/broadcastprofile.cpp:80]:
```suggestion
const QString kDefaultStreamName = QStringLiteral("Mixxx");
```
As @Holzhaus said: Mixxx is not going to be translated, you can use QStringLiteral instead.

**Review by ronso0 — CHANGES_REQUESTED** [review]

**Comment by ronso0** [comment]
This is a harmless bug fix so you can target mixxx:2.3 instead of master.

There shouldn't be a need to rebase, but in case you see unrelated commits poping up here after you switched to mixxx:2.3, just run `git rebase -i XXX/2.3` where XXX is what you named the mixxxdj/mixxx remote.

**Comment by ronso0** [comment]
Unfortunately users that upgraded Mixxx can't benefit from this fix instantly due to the following constraints (not caused by this branch):
1. I expected "Create new connection" in Pref > Broadcast to create a new connection from the default profile, but it just copies the existing connection. That was discussed only briefly in https://github.com/mixxxdj/mixxx/pull/1300#issuecomment-317378535.
As @Palakis mentioned, we shoul dhave `Clone (selected) connection` and `Create new (default) connection`
2. Reset to defaults does nothing (on purpose), but this too inhibits getting a fresh default connection
3. last connection cannot be deleted

All in all, users can not get a default connection again after altering the initial settings unless they remove all connection files from `$CONFIG_DIR/broadcast_profiles` manually.

File: src/preferences/broadcastprofile.cpp (hunk 1 above)
Comment (PawBud) [reply src/preferences/broadcastprofile.cpp]: I see so that's what the QStringLiteral meant!

**Comment by PawBud** [comment]
> Unfortunately users that upgraded Mixxx can't benefit from this fix instantly due to the following constraints (not caused by this branch):
> 
> 1. I expected "Create new connection" in Pref > Broadcast to create a new connection from the default profile, but it just copies the existing connection. That was discussed only briefly in [#1300 (comment)](https://github.com/mixxxdj/mixxx/pull/1300#issuecomment-317378535).
>    As @Palakis mentioned, we shoul dhave `Clone (selected) connection` and `Create new (default) connection`
> 2. Reset to defaults does nothing (on purpose), but this too inhibits getting a fresh default connection
> 3. last connection cannot be deleted
> 
> All in all, users can not get a default connection again after altering the initial settings unless they remove all connection files from `$CONFIG_DIR/broadcast_profiles` manually.
> 
> **EDIT**
> 3. was already touch in #2158 bu not considered worth fixing

So, bottom line, do I have to wait until a Fix is issued, although you clearly stated that they r not really "bugs"

**Comment by ronso0** [comment]
never mind, as I wrote it's unrelated to your fix, just a heads-up for everyone reading this.
Users starting with a new Mixxx install (or clean config) will get the stream name filled.

> This is a harmless bug fix so you can target mixxx:2.3 instead of master.

This can be done at the top of this page, 'Edit', then select mixxx/2.3 in the drop-down.
 
> There shouldn't be a need to rebase, but in case you see unrelated commits poping up here after you switched to mixxx:2.3, just run `git rebase -i XXX/2.3` where XXX is what you named the mixxxdj/mixxx remote.

**Comment by PawBud** [comment]
@ronso0 so the thing is I picked the commits (and didn't squash any), It should rebase but it's showing some conflicts, I tried to manually `Accept Incoming Changes` and merged them with the commit message as `Resolving Conflicts` in VSC, but it just kept on aborting. What am I doing wrong?

**Comment by Holzhaus** [comment]
Please follow the instructions here: https://github.com/mixxxdj/mixxx/wiki/Using%20Git#targeting-another-base-branch

**Comment by PawBud** [comment]
> Please follow the instructions here: https://github.com/mixxxdj/mixxx/wiki/Using%20Git#targeting-another-base-branch

just what I needed! Thanks a lot, I ll remember the command from now on I guess :).

**Comment by PawBud** [comment]
@ronso0, It's weird that the default string is not showing up:-
Screenshot from 2020-08-28 19-28-20

**Comment by ronso0** [comment]
I guess that's exactly what I discovered [above](https://github.com/mixxxdj/mixxx/pull/3043#issuecomment-679268439)

Try removing all files from `$CONFIG_DIR/broadcast_profiles` (~/.mixxx/broadc.. on linux), or remove/backup the config dir entirely.
Then it shows up for the initial default new connection.

**Comment by PawBud** [comment]
> I guess that's exactly what I discovered [above](https://github.com/mixxxdj/mixxx/pull/3043#issuecomment-679268439)
> 
> Try removing all files from `$CONFIG_DIR/broadcast_profiles` (~/.mixxx/broadc.. on linux), or remove/backup the config dir entirely.
> Then it shows up for the initial default new connection.

I can't really find the config directory, there is a file `mixxx\.tx\config`

**Comment by ronso0** [comment]
https://github.com/mixxxdj/mixxx/wiki/Finding%20the%20Mixxx.log%20file

**Comment by PawBud** [comment]
Screenshot from 2020-08-28 20-15-44
Screenshot from 2020-08-28 20-15-36
It's pretty embarrassing to further ask your help, but I can't find any file named `mixxx.log`, moreover, I have verified that there are no Hidden dir.

**Comment by Holzhaus** [comment]
The hidden `.mixxx` directory is in your home directory.

**Comment by ronso0** [comment]
Please read [Wiki: Finding the mixxx.log file](https://github.com/mixxxdj/mixxx/wiki/Finding%20the%20Mixxx.log%20file)
There's a `.mixxx` folder in you Home directory, like `home/YourUsername/.mixxx/`
Make sure you have 'Show hidden folder' enabled for your file manager.

**Comment by ronso0** [comment]
(I update the wiki page to include that info)

**Comment by PawBud** [comment]
oh ok thanks a lot, should be easier for others now :)

**Comment by ronso0** [comment]
Any comments on the minimal changes? If not I'd merge this.

**Comment by daschuer** [comment]
@PawBud: Thank you for this fist pull request.

Before merge we need you permission. 
Please sign: 
https://docs.google.com/a/mixxx.org/spreadsheet/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ
and comment here when done.

**Review by daschuer — APPROVED** [review]
LGTM.  (the CI failure is unrelated.)

**Comment by PawBud** [comment]
Done @daschuer

**Comment by daschuer** [comment]
Thank you.

**Comment by daschuer** [comment]
https://github.com/mixxxdj/mixxx/commit/6c5bc4d43d2

**Pull request merged**